### PR TITLE
[index] Use a #pragma to apply module namespacing for exported ObjC header and match Swift namespaced USRs with the clang side

### PR DESCRIPTION
**Pull request by akyrtzi**

### CODE REVIEW

**Comment by akyrtzi** [comment]
@swift-ci smoke test

**Comment by akyrtzi** [comment]
@swift-ci smoke test

**Comment by akyrtzi** [comment]
@jrose-apple , could you review the PrintAsObjC.cpp part ?
@nathawes , please take a look at the rest.

**Comment by akyrtzi** [comment]
@swift-ci smoke test

**Comment by akyrtzi** [comment]
@swift-ci smoke test

File: lib/PrintAsObjC/PrintAsObjC.cpp [hunk 1]
```diff
@@ -2682,8 +2701,15 @@ class ModuleWriter {
     out <<
         "#pragma clang diagnostic ignored \"-Wproperty-attribute-mismatch\"\n"
         "#pragma clang diagnostic ignored \"-Wduplicate-method-arg\"\n"
+        "#if __has_warning(\"-Wpragma-clang-attribute\")\n"
```
Comment (jrose-apple) [inline lib/PrintAsObjC/PrintAsObjC.cpp]: Doesn't "pragma clang diagnostic ignored" already ignore unknown warnings?

**Review by jrose-apple — COMMENTED** [review]
PrintAsObjC LGTM, other than one comment.

File: lib/PrintAsObjC/PrintAsObjC.cpp (hunk 1 above)
Comment (akyrtzi) [inline lib/PrintAsObjC/PrintAsObjC.cpp]: No, you get `warning: unknown warning group`

Comment (jrose-apple) [inline lib/PrintAsObjC/PrintAsObjC.cpp]: Hm, okay.

**Pull request merged**